src/main/java/seedu/address/model/FoodBook.java

```diff
@@ -0,0 +1,121 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.food.Food;
+import seedu.address.model.food.UniqueFoodList;
+
+
+/**
+ * Wraps all data at the address-book level
+ * Duplicates are not allowed (by .isSameFood comparison)
+ */
+public class FoodBook implements ReadOnlyFoodBook {
+
+    private final UniqueFoodList foods;
+
+    /*
+     * The 'unusual' code block below is a non-static initialization block, sometimes used to avoid duplication
+     * between constructors. See https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html
+     *
+     * Note that non-static init blocks are not recommended to use. There are other ways to avoid duplication
+     *   among constructors.
+     */
+    {
+        foods = new UniqueFoodList();
+    }
```

Comment:
Maybe you could change this non-static initialization block to other ways such as [What is the alternative to a non-static initialization block?](https://stackoverflow.com/questions/18096696). The comments on top of this also suggest that non-static init blocks are not recommended to use.

Reply:
Understood, resolved.